### PR TITLE
Team velocity report

### PR DESCRIPTION
**Pull request by tehreemtalat7**

### CODE REVIEW

File: app/helpers/employees_helper.rb [hunk 1]
```diff
@@ -19,4 +19,8 @@ def employee_velocity_nan?(employee_velocity)
       return employee_velocity
     end
   end
+
+  def team_name(team_id)
+    Team.find(team_id).name
```
Comment (mabdullahkhalil) [inline app/helpers/employees_helper.rb]: current_employee.company.teams.find(team_id).name

File: app/models/employee.rb [hunk 2]
```diff
@@ -62,17 +62,27 @@ def time_spent_on_tasks(employee_id, employee_tasks)
...
-      @employee_tasks_data[employee.id]['total_tasks'] = @employee_tasks.count 
-      @employee_tasks_data[employee.id]['total_time'] = time_spent_on_tasks(employee.id, @employee_tasks)
-      @employee_tasks_data[employee.id]['total_complexity'] = compute_total_complexity_of_tasks(@employee_tasks)
+
+    @employee_teams.each do |team|
+      @employees_list = team.employees
+      @employee_tasks_data[team.id] = {}
+      @employees_list.each do |employee|
+        @employee_tasks_data[team.id][employee.id] = {}
+        @employee_tasks_data[team.id][employee.id]['employee_id'] = employee.id
```
Comment (mabdullahkhalil) [inline app/models/employee.rb]: the way you are defining hash is wrong. 
It should be something like this.
`@employee_tasks_data[team.id][employee.id] = {
employee_id: employee.id,
employee_name: employee.name,
velocity: employee.compute_one_employee_velocity(employee.id, @employee_tasks)
}`

File: app/helpers/employees_helper.rb (hunk 1 above)
Comment (tehreemtalat7) [reply app/helpers/employees_helper.rb]: done

File: app/models/employee.rb (hunk 2 above)
Comment (tehreemtalat7) [reply app/models/employee.rb]: changed!

File: Gemfile [hunk 3]
```diff
@@ -23,6 +23,8 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'select2-rails'
 gem 'sequenceid'
 gem 'uglifier', '>= 1.3.0'
+gem 'wicked_pdf'
```
Comment (zainsaleem4338) [inline Gemfile]: version

File: Gemfile [hunk 4]
```diff
@@ -23,6 +23,8 @@ gem 'sdoc', '~> 0.4.0', group: :doc
...
+gem 'wkhtmltopdf-binary'
```
Comment (zainsaleem4338) [inline Gemfile]: version

File: app/assets/stylesheets/custom.css.scss [hunk 5]
```diff
@@ -72,10 +72,21 @@ a:focus {
   border-top: 1px solid #eaecf4;
 }
 
+.velocity-table {
+	margin-bottom: 50px;
```
Comment (zainsaleem4338) [inline app/assets/stylesheets/custom.css.scss]: spacing issue

File: app/helpers/employees_helper.rb [hunk 6]
```diff
@@ -1,4 +1,26 @@
-require 'attendance'
-
 module EmployeesHelper
+  def not_team_task?(task)
+    !task.assignable_type.eql? 'Team'
+  end
+
+  def team_task?(task)
+    return false if task.eql? -1
```
Comment (zainsaleem4338) [inline app/helpers/employees_helper.rb]: what's '-1', should be a constant

File: app/helpers/employees_helper.rb [hunk 7]
```diff
@@ -1,4 +1,26 @@
...
+    task.assignable_type.eql? 'Team'
+  end
+
+  def no_current_task?(task)
+    task.eql? -1
```
Comment (zainsaleem4338) [inline app/helpers/employees_helper.rb]: and here

File: app/helpers/employees_helper.rb (hunk 6 above)
Comment (tehreemtalat7) [reply app/helpers/employees_helper.rb]: was using it before with previous logic, now its not used, i'll remove them

File: app/views/tasks/employee_tasks.html.erb [hunk 8]
```diff
@@ -41,17 +26,23 @@
...
+      </td>
       <td>
         <%= task.expected_end_date.to_formatted_s(:short) %>
       </td>
       <td>
-        <%= form_for task, url: update_task_logtime_path(task), remote: true, html: { method: :patch } do |form|%>
+        <%= task_logtime(task) %>
+      </td>
+      <td>
+        <%= form_for task.task_time_logs.where(employee_id: current_employee.id).first, url: update_task_logtime_path(task), remote: true, html: { method: :patch } do |form|%>
```
Comment (mabdullahkhalil) [inline app/views/tasks/employee_tasks.html.erb]: do this in the controller 
`task.task_time_logs.where(employee_id: current_employee.id).first`

Comment (tehreemtalat7) [reply app/views/tasks/employee_tasks.html.erb]: umm i have to do it for every task, can't just pass one param here

**Pull request merged**